src/searchengine/nova/engines/btdb.py

```diff
@@ -0,0 +1,147 @@
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+from HTMLParser import HTMLParser
+#qBt
+from novaprinter import prettyPrinter
+from helpers import download_file, retrieve_url
+
+class btdb(object):
+    """ Search engine class """
+    url = 'https://btdb.in'
+    name = 'BTDB'
+
+    def download_torrent(self, info):
+        """ Downloader """
+        print(download_file(info))
+
+    class MyHtmlParser(HTMLParser):
+        """ Parser class """
+        def __init__(self, results, url):
+            HTMLParser.__init__(self)
+            self.results = results
+            self.url = url
+            self.current_item = {} # One torrent result
+            self.add_query = True
+            self.torrent_info_index = 0 # Count of the meta data encountered
+            self.torrent_info_array = []
+            self.meta_data_grabbing = 0
+            self.meta_data_array = []
+            self.torrent_no_files = 0
+            self.torrent_date_added = 0
+            self.torrent_popularity = 0
+            self.mangnet_link = ""
+            self.desc_link = ""
+            self.torrent_name = ""
+
+        def handle_starttag(self, tag, attrs):
+            if tag == "span":
+                span_dict = dict(attrs)
+                if "class" in span_dict:
+                    the_class = span_dict["class"]
+                    if the_class == "item-meta-info-value":
+                        self.meta_data_grabbing += 1
+                    else:
+                        self.meta_data_grabbing = 0
+            if tag == "script":
+                return
+            if tag == "li":
+                for attr in attrs:
+                    if attr[1] == "search-ret-item":
+                        self.torrent_info_index = 1
+            if tag == "a":
+                if self.torrent_info_index > 0:
+                    params = dict(attrs)
+                    if "href" in params:
+                        link = params["href"]
+                        if link.startswith("/torrent"):
+                            self.desc_link = "".join((self.url, link))
+                            self.torrent_name = params["title"]
+                        if link.startswith("magnet:"):
+                            self.mangnet_link = link
+
+        def handle_endtag(self, tag): 
+            if tag == "script":
+                return
+            if tag == "div":
+                if self.meta_data_grabbing >0:
```

Comment:
>We don't have rules about python coding style.

PEP8 do check "whitespace around operator" and it's style is close to what we have in cpp.
Maybe python code should adhere to PEP8? (most of it, not all).